docs/howto/query.md

```diff
@@ -0,0 +1,227 @@
+(query)=
+# How to query the database
+
+## Create/Make your types usable by SQLair
+The first step in building your query is to decide what you would like to put
+into/get out of the database. Identify the types in your Go program that hold
+this information or create them if they don't already exist.
+
+Make sure that the types are in the right format to work with SQLair.
+
+For example:
+```go
+type Employee struct {
+    ID         string `db:"id"`
+    Name       string `db:"name"`
+    LocationID string `db:"location_id"`
+}
+
+type Location struct {
+    ID   string `db:"id"`
+    City string `db:"city"`
+}
+```
+
+> See more: {ref}`types` 
+
+## Write your query string using the types
+
+SQLair queries are regular SQL queries with special input and output expressions
+inserted which allow you to reference the Go types directly from the SQL.
+
+Write your SQL query with SQLair input expressions instead of query parameters
+and SQLair output expressions instead of columns to select.
+
+For example:
+```
+query := `
+    SELECT &Employee.*
+    FROM employee
+    WHERE location_id = $Location.id
+`
+```
+
+> See more {ref}`input-expression-syntax`, {ref}`output-expression-syntax`.
+
+## Prepare your query string to get a Statement
+
+Now you have your query string, you need to pass it to `sqlair.Prepare` along
```

Comment:
I'm not too sure about this, I think in python people do that a lot, but in Go less so. To me that syntax would specifically imply it was a function with no arguments that returned nothing. I could list out what its arguments and return types but then it would be too long, and that belongs in reference anyway.

Reply:
OK. By all means, let's do what's standard in Go.